frontend/taipy-gui/base/src/cookieHandler.ts

```diff
@@ -0,0 +1,66 @@
+import { Socket } from "socket.io-client";
+import { TaipyApp } from "./app";
+import { initSocket } from "./socket";
+import axios from "axios";
+import { getLocalStorageValue } from "../../src/context/utils";
+
+export const TAIPY_RESOURCE_HANDLER = "tprh";
+
+declare global {
+    interface Window {
+        deleteCookie: () => void;
+    }
+}
+
+export class CookieHandler {
+    resourceHandlerId: string;
+    constructor() {
+        this.resourceHandlerId = getLocalStorageValue(TAIPY_RESOURCE_HANDLER, "");
+    }
+    async init(socket: Socket, taipyApp: TaipyApp) {
+        const hasValidCookies = await this.verifyCookieStatus();
+        if (!hasValidCookies) {
+            await this.deleteCookie();
+            localStorage.removeItem(TAIPY_RESOURCE_HANDLER);
+            window.location.reload();
+            return;
+        }
+        this.addBeforeUnloadListener();
+        initSocket(socket, taipyApp);
+    }
+
+    async verifyCookieStatus(): Promise<boolean> {
+        // check to see if local storage has the resource handler id (potentially having a cookie)
+        // If not, then some part of the code must have removed the cookie
+        // or wants to remove the cookie by removing the local storage
+        if (!this.resourceHandlerId) {
+            return new Promise((resolve) => resolve(false));
+        }
+        try {
+            // call to get cookie status
+            const { data } = await axios.get("taipy-rh");
+            // validate cookie status
+            if (!data || !data.rh_id || data.rh_id !== this.resourceHandlerId) {
+                return new Promise((resolve) => resolve(false));
+            }
+        } catch (error) {
+            console.error("Error while validating cookie:", error);
+            return new Promise((resolve) => resolve(false));
+        }
+        return new Promise((resolve) => resolve(true));
+    }
+
+    addBeforeUnloadListener() {
+        window.deleteCookie = () => {
+            axios.delete("taipy-rh");
+        };
+        window.addEventListener("beforeunload", () => {
+            localStorage.removeItem(TAIPY_RESOURCE_HANDLER);
+            window.deleteCookie();
```

Comment:
why not use `self.deleteCookie()`
with `const self = this;` before the addEventListener

Reply:
this.deleteCookie().bind(this) should do the trick but I just want to make sure that the window does not have any dependency on the teardown function